projects/plugins/boost/app/assets/src/js/features/critical-css/lib/stores/critical-css-state.ts

```diff
@@ -20,7 +20,12 @@ export const criticalCssState = {
 	subscribe: cssStateStore.subscribe,
 	refresh: async () => {
 		const status = await stateClient.endpoint.GET();
-		if ( status ) {
+		const storeStatus = get( cssStateStore );
+
+		// This is a temporary fix.
+		// Compare status and storeStatus by serializing and update store if they differ.
+		// This is to avoid unnecessary updates to the store, which can cause rerenders.
+		if ( JSON.stringify( status ) !== JSON.stringify( storeStatus ) ) {
```

Comment:
I don't like this code, but I'm letting it pass for now as this file will be refactored in the next phase of this :)